File: src/main/cpp/lib/SwerveWheelSim.cpp
#include <lib/SwerveWheelSim.h>

SwerveWheelSim::SwerveWheelSim(
        frc::LinearSystem<1, 1, 1> plant, 
        frc::DCMotor motor, 
        double gearboxRatio, 
        units::meter_t wheel_radius, 
        const std::array<double, 1>& measurementStdDevs):
    FlywheelSim(plant, motor, gearboxRatio, measurementStdDevs),
    wheel_radius(wheel_radius)
{}

units::meters_per_second_t SwerveWheelSim::GetLinearVelocity() const {
    return wheel_radius * GetAngularVelocity() / 1_rad;
}

File: src/main/java/io/robojackets/sim/SwerveWheelSim.java
package io.robojackets.sim;

import edu.wpi.first.math.Matrix;
import edu.wpi.first.math.numbers.N1;
import edu.wpi.first.math.system.LinearSystem;
import edu.wpi.first.math.system.plant.DCMotor;
import edu.wpi.first.wpilibj.simulation.FlywheelSim;

public class SwerveWheelSim extends FlywheelSim {
  double wheel_radius;

  public SwerveWheelSim(
      LinearSystem<N1, N1, N1> plant,
      DCMotor gearbox,
      double gearing,
      double wheel_radius,
      Matrix<N1, N1> measurementStdDevs) {
    super(plant, gearbox, gearing, measurementStdDevs);
    this.wheel_radius = wheel_radius;
  }

  public double GetLinearVelocity() {
    return getAngularVelocityRadPerSec() * this.wheel_radius;
  }
}
